include/nbl/system/IFileArchive.h

```diff
@@ -64,10 +64,10 @@ class IFileArchive : public core::IReferenceCounted
 		};
 
```

Comment:
ok we could rearrange this a bit
```cpp
struct SFileList
{
        struct SEntry
        {
            //same stuff as `SListEntry` right now
        };
        using refctd_storage_t = std::shared_ptr<const std::vector<SEntry>>;
        using range_t = core::SRange<const SEntry>;
        
        inline range_t  operator range_t() const {return m_range;}

        SFileList(const SFileList&) = default;
        SFileList(SFileList&&) = default;
        SFileList& operator=(const SFileList&) = default;
        SFileList& operator=(SFileList&&) = default;

    private:
        // default ctor full range
        SFileList(refctd_storage_t _data) : m_data(_data), m_range({_data.data(),_data.data()+_data.size()}) {}
    
        friend class IFileArchive;
        refctd_storage_t m_data;
        range_t m_range;
};
```

then you could do
```
virtual inline SFileList listAssets() const
{
   return SFileList(m_items.load());
}
virtual inline SFileList listAssets(const path& asset_path) const
{
    auto trimmedList = listAssets();
    {
       auto begin = trimmedList.m_data->begin();
       auto end = trimmedList.m_data->end();
       const SFileList::SEntry itemToFind = { pathRelativeToArchive };
       trimmedList.m_range = {&(*std::lower_bound(begin,end,itemToFind)),&(*std::upper_bound(begin,end,itemToFind))};
    }
    return trimmedList;
}
```
and a keep an `m_items` as a 
```cpp
std::atomic<refctd_storage_t> m_items;
```

Reply:
@Hazardu